src/jit/lower.cpp

```diff
@@ -3685,6 +3685,9 @@ void Lowering::DoPhase()
         {
             TreeNodeInfoInit(node);
 
+            // Only nodes that produce values should have a non-zero dstCount.
+            assert((node->gtLsraInfo.dstCount == 0) || node->IsValue());
+
```

Comment:
We can't assert that a node that is a value has a non-zero `dstCount` due to contained nodes: such nodes must be values (IIUC) and must have a `dstCount` and `srcCount` of 0.